test/e2e/specs/site-editor/font-library.spec.js

```diff
@@ -86,4 +86,161 @@ test.describe( 'Font Library', () => {
 			).toBeVisible();
 		} );
 	} );
+
+	test.describe( 'When user has permissions to edit font families', () => {
+		test.beforeAll( async ( { requestUtils } ) => {
+			await requestUtils.activateTheme( 'twentytwentythree' );
+		} );
+
+		test.beforeEach( async ( { admin, editor, page } ) => {
+			await admin.visitSiteEditor();
+			await editor.canvas.locator( 'body' ).click();
+			await page.getByRole( 'button', { name: /styles/i } ).click();
+			await page
+				.getByRole( 'button', { name: /typography styles/i } )
+				.click();
+			await page
+				.getByRole( 'button', {
+					name: /manage fonts/i,
+				} )
+				.click();
+		} );
+
+		test( 'should display the "Upload" tab and upload a local font', async ( {
+			page,
+		} ) => {
+			await expect(
+				page.getByRole( 'tab', { name: /upload/i } )
+			).toBeVisible( { timeout: 40000 } );
+
+			// Ensure font does not exist before uploading
+			if (
+				await page
+					.getByRole( 'button', { name: /commissioner/i } )
+					.isVisible( { timeout: 40000 } )
+			) {
+				await page
+					.getByRole( 'button', { name: /commissioner/i } )
+					.click();
+				await page.getByRole( 'button', { name: /delete/i } ).click();
+				await page.getByRole( 'button', { name: /delete/i } ).click();
+			}
```

Comment:
Can this be achievable programmatically before the test is run?

Reply:
Do you mean by doing something like clearing the installed fonts via the REST API, rather than via the UI in the test?